src/main/java/org/truffleruby/language/LazyRubyNode.java

```diff
@@ -62,4 +61,23 @@ public RubyNode resolve() {
         });
     }
 
+    private RubyNode getOrCreateMasterResolution() {
+        RubyNode masterResolution = resolutionMaster[0];
+        if (masterResolution == null) {
```

Comment:
I think this pattern would request `resolutionMaster` to be `volatile`

Reply:
Which it can't be since it's an array... I think it will have to be an `AtomicReference`.